dali/graphics-api/graphics-api-accessor.h

```diff
@@ -0,0 +1,72 @@
+#ifndef DALI_GRAPHICS_API_ACCESSOR_H
+#define DALI_GRAPHICS_API_ACCESSOR_H
+/*
+ * Copyright (c) 2017 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// EXTERNAL INCLUDES
+#include <memory>
+
+#include <dali/graphics/graphics-object-owner.h>
+
+namespace Dali
+{
+namespace Graphics
+{
+namespace API
+{
+
+template< typename T >
+class Accessor
+{
+public:
+  Accessor(ObjectOwner< T >& owner, typename ObjectOwner< T >::Handle handle)
+  : mOwner(owner),  // save owner to access object
+    mHandle(handle) // handle to the object
+  {
+  }
+
+  operator bool() const
+  {
+    return mOwner.Contains(mHandle);
+  }
+
+  auto& Get()
```

Comment:
isn't that c++14 only? wouldn't it be safer, for now, to stick with code compatible with c++11 if we don't know yet what standard we are going to use?